x/evm/keeper/vote_handler.go

```diff
@@ -0,0 +1,162 @@
+		}
+
+		if len(errors) != 0 {
+			for _, err := range errors {
+				keeper.Logger(ctx).Error(fmt.Sprintf("failed to process event: %s", err.Error()))
+			}
+			return fmt.Errorf("failed to process events: %s", errors)
+		}
+
+		return nil
+	}
+}
+
+func handleVoteConfirmDeposit(ctx sdk.Context, k types.BaseKeeper, n types.Nexus, chain nexus.Chain, event types.Event) error {
+
+	keeper := k.ForChain(chain.Name)
+	transferEvent := event.GetEvent().(*types.Event_Transfer)
+
+	k.Logger(ctx).Info(fmt.Sprintf("deposit confirmation result is %s", transferEvent.Transfer.String()), "chain", chain.Name)
+
+	// get deposit address
+	burnerInfo := keeper.GetBurnerInfo(ctx, transferEvent.Transfer.To)
+	if burnerInfo == nil {
+		return fmt.Errorf("no burner info found for address %s", transferEvent.Transfer.To.Hex())
+	}
+
+	depositAddr := nexus.CrossChainAddress{Chain: chain, Address: transferEvent.Transfer.To.Hex()}
+	recipient, ok := n.GetRecipient(ctx, depositAddr)
+	if !ok {
+		return fmt.Errorf("cross-chain sender has no recipient")
+	}
+
+	amount := sdk.NewCoin(burnerInfo.Asset, sdk.NewIntFromBigInt(transferEvent.Transfer.Amount.BigInt()))
+	transferID, err := n.EnqueueForTransfer(ctx, depositAddr, amount)
+	if err != nil {
+		return err
+	}
+
+	k.Logger(ctx).Info(fmt.Sprintf("%s deposit confirmation result to %s", chain.Name, burnerInfo.BurnerAddress))
+
+	// handle poll result
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(types.EventTypeDepositConfirmation,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(types.AttributeKeyChain, chain.Name),
+			sdk.NewAttribute(types.AttributeKeySourceChain, chain.Name),
+			sdk.NewAttribute(types.AttributeKeyDestinationChain, recipient.Chain.Name),
+			sdk.NewAttribute(types.AttributeKeyDestinationAddress, recipient.Address),
+			sdk.NewAttribute(types.AttributeKeyAmount, transferEvent.Transfer.Amount.String()),
+			sdk.NewAttribute(types.AttributeKeyAsset, burnerInfo.Asset),
+			sdk.NewAttribute(types.AttributeKeyDepositAddress, depositAddr.Address),
+			sdk.NewAttribute(types.AttributeKeyTokenAddress, burnerInfo.TokenAddress.Hex()),
+			sdk.NewAttribute(types.AttributeKeyTxID, event.TxId.Hex()),
+			sdk.NewAttribute(types.AttributeKeyTransferID, transferID.String()),
+		))
+
+	return nil
+}
+
+func handleVoteConfirmToken(ctx sdk.Context, k types.BaseKeeper, chain nexus.Chain, event types.Event) error {
+	keeper := k.ForChain(chain.Name)
+	tokenDeployedEvent := event.GetEvent().(*types.Event_TokenDeployed)
+
+	token := keeper.GetERC20TokenBySymbol(ctx, tokenDeployedEvent.TokenDeployed.Symbol)
+	if token.Is(types.NonExistent) {
+		return fmt.Errorf("token %s does not exist", tokenDeployedEvent.TokenDeployed.Symbol)
+	}
+
+	if token.GetAddress() != tokenDeployedEvent.TokenDeployed.TokenAddress {
+		return fmt.Errorf("token address %s does not match expected %s", tokenDeployedEvent.TokenDeployed.TokenAddress.Hex(), token.GetAddress().Hex())
+	}
+
```

Comment:
What about `RejectDeployment` if the vote fails, so it can be retried. Otherwise if someone confirms token deployment before confirmation height is reached, this wouldn't allow us to retrigger it, right?

Reply:
Does this mean we need a poll failure handler, revert state changes in the module when a poll failed?